### ISSUE TITLE
🐞 - `ComboBox` throws errors `TypeError` & `ExpressionChangedAfterItHasBeenCheckedError`

### ISSUE THREAD

**Issue by nsbarsukov** [Tinkoff/taiga-ui#2729]
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/combobox-console-errors?file=src/app/app.component.html

### Is this issue blocking you?

Non-Blocking

## Reproduction
1. Open https://stackblitz.com/edit/combobox-console-errors?file=src/app/app.component.html
2. Click on the combobox

<img width="1791" alt="combobox-errors" src="https://user-images.githubusercontent.com/35179038/190974255-31ee5932-679d-4675-ba55-9f2c0710e044.png">

## Error logs
<details>
<summary>See full logs of <strong>TypeError</strong></summary>

```console
ERROR TypeError: Class constructor ElementRef cannot be invoked without 'new'
    at new TuiElementDirective (element.directive.ts:9:9)
    at NodeInjectorFactory.TuiElementDirective_Factory [as factory] (ɵfac.js? [sm]:1:1)
    at getNodeInjectable (di.ts:636:15)
    at instantiateAllDirectives (shared.ts:1229:7)
    at createDirectivesInstances (shared.ts:556:2)
    at ɵɵelementStart (element.ts:152:3)
    at TuiDataListWrapperComponent_tui_data_list_0_button_1_Template (ɵcmp.js? [sm]:16:13)
    at executeTemplate (shared.ts:522:4)
    at renderView (shared.ts:326:6)
    at TemplateRef.createEmbeddedView (view_container_ref.ts:271:12)
```
</details>

<details>
<summary>See full logs of <strong>ExpressionChangedAfterItHasBeenCheckedError</strong></summary>

```console
ERROR Error: NG0100: ExpressionChangedAfterItHasBeenCheckedError:
Expression has changed after it was checked.
Previous value: 'undefined'.
Current value: 'listbox'.
It seems like the view has been created after its parent and its children have been dirty checked.
Has it been created in a change detection hook? Find more at https://angular.io/errors/NG0100
    at throwErrorIfNoChangesMode (errors.ts:110:5)
    at bindingUpdated (bindings.ts:85:9)
    at ɵɵattribute (interpolation.ts:18:1)
    at TuiDataListComponent_HostBindings (ɵcmp.js:0:0)
    at processHostBindingOpCodes (shared.ts:101:9)
    at refreshView (shared.ts:447:8)
    at refreshEmbeddedViews (shared.ts:1595:58)
    at refreshView (shared.ts:421:6)
    at refreshComponent (shared.ts:1646:11)
    at refreshChildComponents (shared.ts:138:3)


[...]

ERROR Error: NG0100: ExpressionChangedAfterItHasBeenCheckedError:
Expression has changed after it was checked.
Previous value for 'overscroll-behavior': 'undefined'.
Current value: 'contain'.
It seems like the view has been created after its parent and its children have been dirty checked.
Has it been created in a change detection hook? Find more at https://angular.io/errors/NG0100
    at throwErrorIfNoChangesMode (errors.ts:110:5)
    at bindingUpdated (bindings.ts:85:9)
    at checkStylingProperty (styling.ts:229:34)
    at ɵɵstyleProp (styling.ts:97:4)
    at TuiOverscrollDirective_HostBindings (ɵfac.js? [sm]:1:1)
    at processHostBindingOpCodes (shared.ts:101:9)
    at refreshView (shared.ts:447:8)
    at refreshComponent (shared.ts:1646:11)
    at refreshChildComponents (shared.ts:138:3)
    at refreshView (shared.ts:449:33) 
```
</details>

### Angular version

14

### Taiga UI version

3.3.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

**Comment by waterplea** [comment]
Can't seem to reproduce it on exact same code. Seems like some kind of StackBlitz only issue.

**Comment by expdts** [comment]
I get the same error, but only in the dialog box. 
[Example](https://github.com/expdts/taiga-tui-combo-box)

**Comment by expdts** [comment]
Please correct the error. Because of it, it is not possible to upgrade to version 3.

**Comment by nsbarsukov** [comment]
@expdts thanks for problem's reproduction!

we will investigate this issue again.

**Comment by vladimirpotekhin** [comment]
Hi @expdts 
while we're investigating the problem, you can move the variable declaration under the `<ng-template>` (eg ng-container). This should solve your problem until we find another solution
<img width="715" alt="Снимок экрана 2022-11-18 в 17 18 02" src="https://user-images.githubusercontent.com/46284632/202714998-9a8e30eb-a8cc-4430-896a-da8524fd64ef.png">

**Comment by expdts** [comment]
@vladimirpotekhin
Can you provide a complete example of a temporary solution? I can't avoid making a mistake.

**Comment by vladimirpotekhin** [comment]
<img width="1363" alt="Снимок экрана 2022-12-01 в 15 21 33" src="https://user-images.githubusercontent.com/46284632/205051680-2c4fcbbb-1ea4-4272-8741-02b12410ffd5.png">

@expdts after this change there are no errors in the repo you mentioned above

**Comment by waterplea** [comment]
Fixed in #3200 expected to come out today.
